modules/core/site.js

```diff
@@ -1783,7 +1783,18 @@ $(function() {
         $('.reset_default_value_select').on("click", reset_default_value_select);
         $('.reset_default_value_input').on("click", reset_default_value_input);
     }
-    
+
+    $(window).on('beforeunload', function(event) {
+        if ($('#unsaved_changes').val() > 0) {
```

Comment:
Unsaved changes input is updated to 1 only in specific cases in imap/smtp/pop3 modules and a few other special places. It is not updated when user manually changes settings in site settings or profiles, for example. If we count on that setting, we will need to extend all places where settings/forms are updated but not submitted - that would be quite an effort.
Why not use a simpler idea - global JS event handler for form changes and global dirty flag? If flag is raised and there is unsaved data, warn user...

Reply:
@kroky I don't get a better idea on how to handle this. As there are some forms submissions (like the compose form and other) that does not lead to something that user can be saved in user settings.

Reply:
Having a compose form that is half-filled and leaving the page will make user loose the content. I don't think it matters if it is a compose form, a new folder field, a new smtp-server form half-filled or a settings form. I'd vote for a new setting to trigger this behavior system-wide on and off - warn user for unsaved changes.